Change Excessive Retransmit to Excessive Loss (#650)
ietf-tapswg/api-drafts: draft-ietf-taps-interface.md

bidirectional communication to be selected.

> ### Notification of Excessive Loss {#prop-establish-excss-loss-notify}

Reviewer: ```suggestion
### Notification of Loss {#prop-establish-excss-loss-notify}
```

: Ignore

This property specifies whether an application considers it useful to be

> informed in case sent data was lost more often than a certain
> threshold (see {{conn-excss-loss}} for configuration of this threshold).

Author: Okay… I agree on that one – In case we want to keep the semantics of this PR (see comment below) packets of the PR, it hast to say 
```suggestion
informed in case packets of the underlying transport protocol have been lost more often than a certain
threshold (see {{conn-excss-loss}} for configuration of this threshold).
```
or it makes no sense.

> threshold (see {{conn-excss-loss}} for configuration of this threshold).

Reviewer 2: The transport data was not "lost" the IP packets were. I've changed my position to argue against having this metric at all. It is very transport specific. (see my comment in #647).

> threshold (see {{conn-excss-loss}} for configuration of this threshold).

Reviewer: The IP packets are not known to the application. This should really refer to Messages.

enables applications to express their preference for protocols providing a supporting transport feature.

> ### Loss Threshold before Excessive Loss Notification {#conn-excss-loss}

Author: This makes sense for TCP, but not for QUIC with datagrams as @mirjak stated in issue #647 – QUIC still detects the loss, but does not necessarily retransmit the data. Each retransmit is usually also a loss event (ignoring re-ordering problems), but not the other way around.

> ### Loss Threshold before Excessive Loss Notification {#conn-excss-loss}

Reviewer: Right - generally, the threshold doesn't make sense for QUIC, but my other proposed text changes eliminate the threshold's meaning for unreliable data transfer.

> ### Loss Threshold before Excessive Loss Notification {#conn-excss-loss}

Reviewer 3: My point was that having a loss notification does make sense for unreliable transports like datagram in QUIC and therefore it should not only apply for reliable protocols.

> ### Loss Threshold before Excessive Loss Notification {#conn-excss-loss}

Reviewer: Yes! but the threshold makes no sense, and calling it "Excessive" doesn't, either. Keep the loss notification, but use the threshold for reliable transfers only, I say.

Name:
: excessLossNotifyThreshold

Type:
: Numeric, with special value `Disabled`

Default:
: Disabled

> This property specifies the number of lost packtes per second required to cause the application

Reviewer: ```suggestion
For reliable data transfer, this property specifies the number of retransmissions required to cause the application
```

> to be notified of `Excessive Loss`.

Reviewer: ```suggestion
to be notified of `Excessive Retransmission`. In case of unreliable data transfer, configuring this threshold will not have any effect.
```

Connection -> SoftError<>
~~~

> ### Excessive Loss {#conn-excss-loss-notify}

Reviewer: ```suggestion
### Loss {#conn-excss-loss-notify}
```

> This event notifies the application of excessive loss events,

Reviewer: ```suggestion
This event notifies the application of loss events,
```

> e.g., indicated by non-acked packets or exessive retransmissions,

Reviewer: ```suggestion
e.g., indicated by non-acked packets or (in case of reliable data transfer) retransmissions,
```
(note: if you don't accept this suggestion, still fix the typo: s/exessive/excessive)

based on a configured threshold (see {{conn-excss-loss}}).
This will only happen if the underlying protocol stack supports loss detection and, with it, such notifications.

~~~

> Connection -> ExcessiveLoss<>

Reviewer: ```suggestion
Connection -> Loss<>
```

`TCP-specific Property: User Timeout` ({{tcp-uto}}).

* Notification of Excessive Retransmissions (early warning below abortion threshold):

> `Notification of excessive retransmissions` property ({{prop-establish-excss-loss-notify}}).

Reviewer: ```suggestion
`Notification of loss` property ({{prop-establish-excss-loss-notify}}).
```
